cli/cmd/connect.go

```diff
@@ -122,18 +137,53 @@ func isCredentialsURI(str string) bool {
 	// user:password@host:port
 	tcpReStr := `(tcp://)?` + userpassRe + `@([\w\.-]+:\d+)`
 	// unix://user:password@../path
+	// unix://user:password@~/path
 	// unix://user:password@/path
 	// unix://user:password@path
-	unixReStr := `unix://` + userpassRe + `@[./@]*[^\./@]+.*`
-	// user:password@./path
+	unixReStr := `unix://` + userpassRe + `@` + uriPathPrefixRe + `[^\./@]+.*`
+	// user:password@../path
+	// user:password@~/path
 	// user:password@/path
-	pathReStr := userpassRe + `@\.?/[^\./].*`
+	// user:password@./path
+	pathReStr := userpassRe + `@` + systemPathPrefixRe + `[^\./].*`
 
 	uriReStr := "^((" + tcpReStr + ")|(" + unixReStr + ")|(" + pathReStr + "))$"
 	uriRe := regexp.MustCompile(uriReStr)
 	return uriRe.MatchString(str)
 }
 
+// parseBaseURI parses an URI and returns:
+// (network, address)
+func parseBaseURI(URI string) (string, string) {
+	var network, address string
+	URILen := len(URI)
+
+	switch {
+	case URILen > 0 && (URI[0] == '.' || URI[0] == '/' || URI[0] == '~'):
+		network = connector.UnixNetwork
+		address = URI
+	case URILen >= 7 && URI[0:7] == "unix://":
+		network = connector.UnixNetwork
+		address = URI[7:]
+	case URILen >= 6 && URI[0:6] == "tcp://":
+		network = connector.TCPNetwork
+		address = URI[6:]
+	default:
+		network = connector.TCPNetwork
+		address = URI
+	}
+
+	// In the case of a complex uri, shell expansion does not occur, so do it manually.
+	if network == connector.UnixNetwork &&
+		strings.HasPrefix(address, "~/") {
+		if homeDir, err := os.UserHomeDir(); err == nil {
```

Comment:
I'd like to see a warning at least if something goes wrong. Unable to get user home dir in this case. But this one can be a debug type of message, so it will be printed only if verbose mode is enabled.